### PR TITLE
LOG-905: Expanding e2e upgrade tests

### PR DESCRIPTION
**Pull request by ewolinetz**
### Description 
Adding test to increase our CI e2e test coverage: upgrade test from EO version `n-1` to `n` (where `n` is the version specified in the manifest csv).

/cc @blockloop @periklis 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-905

### CODE REVIEW

File: hack/testing-olm-upgrade/test-030-olm-upgrade-n-1-n.sh [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+TIMEOUT_MIN=$((2 * $minute))
+ES_POD_TIMEOUT=$((5 * $minute))
+manifest_dir=${repo_dir}/manifests
+#version="4.5"
+#previous_version="4.4"
+
+discover_versions(){
+  #TODO: move this to an image?
+  curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
```
Comment (periklis) [inline hack/testing-olm-upgrade/test-030-olm-upgrade-n-1-n.sh:29]: IIRC the release repo provides images that include jq `https://github.com/openshift/release/blob/master/core-services/supplemental-ci-images/cli-jq.yaml`. You could do the same with `yq`.

File: hack/testing-olm-upgrade/test-030-olm-upgrade-n-1-n.sh [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+ARTIFACT_DIR=${ARTIFACT_DIR:-"$repo_dir/_output/"}
+ARTIFACT_DIR="${ARTIFACT_DIR}/$(basename ${BASH_SOURCE[0]})"
+if [ ! -d $ARTIFACT_DIR ] ; then
+  mkdir -p $ARTIFACT_DIR
+fi
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+TIMEOUT_MIN=$((2 * $minute))
+ES_POD_TIMEOUT=$((5 * $minute))
+manifest_dir=${repo_dir}/manifests
```
Comment (periklis) [inline hack/testing-olm-upgrade/test-030-olm-upgrade-n-1-n.sh:23]: IIRC, the single-source of truth manifest dir will be `bundle/manifests` in CPAAS right? If yes, we should safeguard ourselves already for that here.

**Review by periklis — COMMENTED** [review]
LGTM.

**Review by periklis — COMMENTED** [review]
What is missing here? Otherwise LGTM

**Review by blockloop — APPROVED** [review]
/lgtm

This is so much better :confetti_ball:

**Pull request merged**